lib/Extension/LanguageServerPhpCsFixer/LanguageServerPhpCsFixerExtension.php

```diff
@@ -11,23 +11,42 @@
 use Phpactor\Extension\LanguageServerPhpCsFixer\LspCommand\FormatCommand;
 use Phpactor\Extension\LanguageServerPhpCsFixer\Model\PhpCsFixerProcess;
 use Phpactor\Extension\LanguageServerPhpCsFixer\Provider\PhpCsFixerDiagnosticsProvider;
+use Phpactor\Extension\LanguageServerPhpCsFixer\VersionResolver\PhpCsFixerVersionResolver;
 use Phpactor\Extension\LanguageServer\Container\DiagnosticProviderTag;
 use Phpactor\Extension\LanguageServer\LanguageServerExtension;
 use Phpactor\Extension\Logger\LoggingExtension;
 use Phpactor\FilePathResolver\PathResolver;
 use Phpactor\LanguageServer\Core\Server\ClientApi;
 use Phpactor\MapResolver\Resolver;
+use Phpactor\VersionResolver\ArbitrarySemVerResolver;
+use Phpactor\VersionResolver\CachedSemVerResolver;
+use Phpactor\VersionResolver\CompoundSemVerResolver;
 
 class LanguageServerPhpCsFixerExtension implements OptionalExtension
 {
+    public const SERVICE_VERSION_RESOLVER = 'php_cs_fixer.version.resolver';
     public const PARAM_PHP_CS_FIXER_BIN = 'language_server_php_cs_fixer.bin';
+    public const PARAM_PHP_CS_FIXER_VERSION = 'language_server_php_cs_fixer.version';
     public const PARAM_ENV = 'language_server_php_cs_fixer.env';
     public const PARAM_SHOW_DIAGNOSTICS = 'language_server_php_cs_fixer.show_diagnostics';
     public const PARAM_CONFIG = 'language_server_php_cs_fixer.config';
     public const PARAM_ENABLED = 'language_server_php_cs_fixer.enabled';
 
     public function load(ContainerBuilder $container): void
     {
+        $container->register(self::SERVICE_VERSION_RESOLVER, function (Container $container) {
+            $pathResolver = $container->expect(FilePathResolverExtension::SERVICE_FILE_PATH_RESOLVER, PathResolver::class);
+
+            $path = $pathResolver->resolve($container->parameter(self::PARAM_PHP_CS_FIXER_BIN)->string());
+
+            return new CachedSemVerResolver(
+                new CompoundSemVerResolver(
+                    new ArbitrarySemVerResolver($container->parameter(self::PARAM_PHP_CS_FIXER_VERSION)->value()),
```

Comment:
can use `->stringOrNull` here instead of `value`

Reply:
Maybe we could add a validator for `array<string,string>` too somewhere separately. For now, I just ignored the environment dictionary like in `PhpCodeSnifferExtension`.